R/pkg/inst/worker/daemon.R

```diff
@@ -30,8 +30,42 @@ port <- as.integer(Sys.getenv("SPARKR_WORKER_PORT"))
 inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
+selectTimeout <- function() {
+  if (is.null(parallel:::selectChildren(timeout = 0))) {
```

Comment:
And ... just in case, these are can be checked via both commands as below:

```
watch -n 0.01 "ps -fe | grep /exec/R"
```

checks forked processes.

```
watch -n 0.01 "lsof -c R | wc -l"
```

check open files (and file descriptors & pipes).